### PR TITLE
feat($thumbnails): add thumnail image placeholders

### LINKED ISSUE #34: Image placeholders
In #32 we began effectively lazyloading image data to speed up the themes site. To do so we removed the `src` attribute from the image tags, which leads to the following perceived render for user agents supporting images with an unprimed cache:

screen shot 2017-04-26 at 11 04 51 am

Because the hyperlink used for individual themes wraps an `img` tag and the image tag itself does not have an intrinsic size assigned (e.g. using width and height attributes or CSS), the click area of the surrounding anchor makes themes unclickable until the images begin to load. This effect is exacerbated when a user scrolls the page rapidly as images far outside the viewport are not preloaded using a lazy queue (possible with [lazysizes](https://github.com/aFarkas/lazysizes), but not really worth it yet).

To improve the current experience for mobile user agents and users on non-cable connections, we can leverage image placeholders, which will appear in place of theme images while theme images are being downloaded. The result of doing so will create a touch/click target the size of the thumbnail image to be loaded. To do this performantly, we can use a base64-endocoded SVG image asset showing the Hugo logo within a rectangle the size of the thumbnail being loaded.

If this is something you'd like to see please send me an SVG asset of the Hugo logo to jhabdas@gmail.com and I'll have a PR ready shortly. The work will be directly portable to the new theme site, so win-win.

### PR DESCRIPTION
**Pull request by ghost**
improve ux with placeholder images while thumbs are downloaded

closes #34

### CODE REVIEW

**Comment by ghost** [comment]
Bud's new theme is so pretty I could no longer look at the issue UX issue I left behind (linked above) when I implemented lazysizes.

This uses a data-encoded SVG images with the HUGO name using Muli bold with a 50% opacity of the link color applied in the theme (pictured in the bottom thumbs below).

**Note:** Haven't tested the blur-up effect here in context of the theme site, but works great on my other sites using traditional LQIP and not static placeholder images as I'm doing in this pull.

themes gohugo io- ipad

File: layouts/partials/head-additions.html [hunk 1]
```diff
@@ -5,9 +5,19 @@
 <link rel="prefetch" href="{{.NextInSection.Permalink }}">
 {{end}}
 
-{{/* TODO: This hides search until the integrated sites are live. It is safe to remove at that point. */}}
 <style media="screen">
+  /* TODO: This hides search until the integrated sites are live. It is safe to remove at that point. */
```
Comment (bep) [inline layouts/partials/head-additions.html]: I'm pretty sure the two approaches to commenting are not equal. I prefer the existing where the comments is not visible in the final source.

Comment (ghost) [reply layouts/partials/head-additions.html]: Will address

**Comment by ghost** [comment]
@budparr time to offer your thoughts on the  thumbnail image placeholder treatment in this pull?

**Comment by budparr** [comment]
I like this idea, @jhabdas I was trying to use animation to solve the same problem, but I like this better.

**Comment by bep** [comment]
@jhabdas if you can fix the "comment" issue, I will merge this.

**Comment by ghost** [comment]
Thanks for taking a look both of you. Updated to remove TODO comments from generated output. For reference the blur up technique [is documented](https://github.com/aFarkas/lazysizes#lqipblurry-image-placeholderblur-up-image-technique) in lazysizes.

**Comment by budparr** [comment]
@jhabdas Looks like you're using "Muli-Bold" but I don't think that's the font we're actually using. Just "Muli" should work though.

**Comment by ghost** [comment]
@budparr I'm on it.

**Comment by budparr** [comment]
At the risk of getting pedantic, the font weight should be `800` instead of `bold` which, I think is `700`. (I tried to limit the downloads, so just used a few weights)

**Comment by ghost** [comment]
No problem. I'm just interested in fixing the UX issue. I perceive the rest as something which could be iterated on if need be. But that's why I wanted to get your input since we have a chance right now to save work later.

I've added another commit to move back to the Regular font style. Sketch doesn't give me the options of applying a font-weight like CSS does, so the new image in commit https://github.com/spf13/hugoThemesSite/pull/43/commits/a551c82145cd6ce0bffecd514e2eece17f97fa9f.

If you'd like something completely different that's fine too. Just let me know.

**Comment by budparr** [comment]
Oh, I see. I was responding to the text in the image string. As long as looks okay (which it does) and not looking up any references that don't exist, it seems perfectly fine.

**Comment by ghost** [comment]
Makes sense. Let me double-check. Just because sketch doesn't offer an option doesn't mean SVG won't handle it. And I can see now `font-family` appears in the code and SVGs may be styled by CSS so let me dig into this one a little more given the font glyphs aren't actually the vectors I'd thought they were.

**Comment by ghost** [comment]
Okay. Made some tweaks to keep more consistent with initial fade-in effect, adjusted font weight to be less bold, added consistency with existing `font-family` style rule used for `sans-serif`  (pictured below in the ghosted - loading - state). Adjusted comment. This feels pretty solid now for a first pass.

screen shot 2017-06-07 at 12 09 45 pm

**Notes:**

- Fade-in animation is not time-based; it will trigger upon observing image load event.
- I tried adding the `class` attribute to the data uri to give more flexibility from existing stylesheets but the CSS would not cascade into the encoded image. We can always update this later if styles change.

**Comment by ghost** [comment]
From my perspective this is merge-ready.

**Comment by ghost** [comment]
Damn that looks sexy in action. 🤘

**Comment by digitalcraftsman** [comment]
Indeed. It's a much better experience to have a placeholder of the same size as the actual thumbnail. Previously, I was annoying if you wanted to click on a thumbnail to checkout a theme but the link always bounces away because once the surrounding thumbnails were loaded they were shown in their expected size.

**Comment by ghost** [comment]
Next feature will be optimized preloading of the `tn.png` based on hover intent.

**Comment by ghost** [comment]
@budparr this new site looks so damn good I have no choice but to retire my github pages site. LOL

**Comment by budparr** [comment]
Thanks for saying so, and doubly thanks for this PR; a great improvement over my animation. You may have seen I do some pre-fetching on prev/next pages, an aggressive strategy, which is to say I think we're on the same page in terms of wringing out any performance we can can. I'm interested to see what you come up with on the images.

**Comment by ghost** [comment]
@budparr awesome. will have to loop back on that later. but if you'd like to give it a stab here's how to eke out more performance (read: a better ux) without any heavy lifting:

- Add `lazyload` class to details page images, change current `src` attribute to `data-src`, set a new `src` attribute equal to the path of the thumbnail and call it a day. This will work extremely well for hunter-gatherers.

**Comment by budparr** [comment]
I don't think lazyload will do much good for what is usually one or two images already in the viewport.

**Comment by ghost** [comment]
> I don't think lazyload will do much good for what is usually one or two images already in the viewport.

What that would do, would be to use the browser-cached thumbnail image while the full-size is downloading, so the user would not have to wait to see any image at all as they moved back-and-fourth between the list pages. The effect would be such that, after the larger image downloaded, it would show the user what the theme looks like on smaller and larger viewports with the same transition we're using on the landing page for the SVG placeholders.

I've got other ideas. This just happens to be the simple one.

**Comment by ghost** [comment]
Try using Chrome Dev Tools and throttling your connection speed to 2 or 3G to see what many users see which many others cannot.

**Comment by budparr** [comment]
I see. But, I'm using pre-fetching, which accomplishes that as well. At any rate, happy to talk more about it, though we should probably do so at the theme repo so everyone else on this thread doesn't keep getting pinged.

**Comment by ghost** [comment]
> I'm using pre-fetching, which accomplishes that as well.

Haven't looked at the implementation. But if we're trying to prefetch many large images on the list pages without using a hover intent it's consuming lots of bandwidth on my data-capped mobile connection right now so I better jump off. LOL

**Comment by budparr** [comment]
Only on prev/next for single pages

**Comment by ghost** [comment]
If your curiosity get the best of you, try my suggestion. I think you'll be surprised. Thanks, Bud.

**Pull request merged**